lib/ASRouterTargeting.jsm

```diff
@@ -502,14 +502,22 @@ this.ASRouterTargeting = {
    * @param {obj} impressions An object containing impressions, where keys are message ids
    * @param {trigger} string A trigger expression if a message for that trigger is desired
    * @param {obj|null} context A FilterExpression context. Defaults to TargetingGetters above.
+   * @param {boolean} returnAll If true, returns a list of all matching messages.
```

Comment:
@piatra what do you think of this? Should I rename method to `findMatchingMessages()`? or?

Reply:
I'm not sure about this one. Personally I would even consider making a new method (even though 99% identical) that fetches all messages that match targeting:
It would make it more obvious when using one or the other and we wouldn't have to pass down `returnAll` through so many methods. Feels less hacky to me.
Not sure, cc @k88hudson ?